korlibs-string/src@apple/korlibs/io/lang/CharsetApple.kt

```diff
@@ -0,0 +1,53 @@
+package korlibs.io.lang
+
+import korlibs.memory.ByteArrayBuilder
+import kotlinx.cinterop.UnsafeNumber
+import platform.Foundation.*
+
+@OptIn(UnsafeNumber::class)
+class CharsetApple(name: String) : Charset(name) {
+    companion object {
+        val encodingMap = mapOf(
+            "UTF-8" to NSUTF8StringEncoding,
+            "ISO-8859-1" to NSISOLatin1StringEncoding,
+            "UTF-16" to NSUTF16StringEncoding,
+            "UTF-16BE" to NSUTF16BigEndianStringEncoding,
+            "UTF-16LE" to NSUTF16LittleEndianStringEncoding,
+            "UTF-32" to NSUTF32StringEncoding,
+            "UTF-32BE" to NSUTF32BigEndianStringEncoding,
+            "UTF-32LE" to NSUTF32LittleEndianStringEncoding,
+            "ASCII" to NSASCIIStringEncoding,
+            "NEXTSTEP" to NSNEXTSTEPStringEncoding,
+            "JAPANESE_EUC" to NSJapaneseEUCStringEncoding,
+            "EUC-JP" to NSJapaneseEUCStringEncoding,
+            "LATIN1" to NSISOLatin1StringEncoding,
+        )
+
+        fun isSupported(name: String) = encodingMap.containsKey(name.uppercase())
+    }
+
+    private val encoding: NSStringEncoding =
+        encodingMap[name.uppercase()] ?: throw IllegalArgumentException("Charset $name is not supported by apple.")
+
```

Comment:
Can we get the NSStringEncoding from a String instead of having a fixed map?

Reply:
@soywiz another option is to replace the map with `when`

Reply:
@soywiz i found some objective-c code for that let me try that.

Reply:
@soywiz i found this working code for that:
```
private val encoding = CFStringConvertEncodingToNSStringEncoding(CFStringConvertIANACharSetNameToEncoding(name.uppercase().toCFStringRef()))
```

```
fun String.toCFStringRef(): CFStringRef {
    return CFBridgingRetain(this as NSString) as CFStringRef
}
```

Let me know what you think. If it looks good, I'll replace the current code with this.

Reply:
Yep, that was the idea 👍

Reply:
Maybe to have in control the lifecycle of  the CFStringRef, something like

```kotlin
inline fun <T> String.useCFStringRef(block: (CFStringRef) -> T): T { 
  val ref = CFBridgingRetain(this as NSString) as CFStringRef
  return block(ref).also { CFRelease(ref) }
}
```

Reply:
That's good idea. I will update it

Reply:
@soywiz Code updated, please check it.